3.2 backports 202308

vadz opened the pull request.

MaartenBent committed:
Fix position of label window in wxStaticBox after DPI change

See #23740.

(cherry picked from commit d847f3b039f469000e4a0673ae1f88a81e889248)

lanurmi committed:
Minor documentation syntax and orthography fixes

See #23759.

(cherry picked from commit 3e4946fc3101befdbdada10662e8b2697a2fd8c3)

swt2c committed:
Support using Numpad Enter for ending wxListCtrl editing (#23762)

See #23761.

(cherry picked from commit 3ae3a4e1f519741132c51e5aa8f83d6f05174501)

carlo-bramini committed:
Fix DynamicLibrary::Load unit test under Cygwin

While running the test_base on CYGWIN, this message will be printed on the console:

-------------------------------------------------------------------------------
DynamicLibrary::Load
-------------------------------------------------------------------------------
/cygdrive/c/Users/carlo/Documents/GitHub/wxWidgets/tests/misc/dynamiclib.cpp:26
...............................................................................

/cygdrive/c/Users/carlo/Documents/GitHub/wxWidgets/tests/misc/dynamiclib.cpp:80: warning:
  Couldn't find libc.so, skipping DynamicLibrary::Load() test.

/cygdrive/c/Users/carlo/Documents/GitHub/wxWidgets/tests/misc/dynamiclib.cpp:84: warning:
  Possible candidates:
  /lib/libc.a
  /usr/lib/libc.a

CYGWIN uses NEWLIB as C library and it is compiled into cygwin1.dll.
This patch adds the right test case for CYGWIN, for letting the execution of the test.

See #23781.

(cherry picked from commit 1bc8631b527603c855d65a9ede45fa3d6e672ea8)

carlo-bramini committed:
Fix wxLocale::GetInfo() unit test under Cygwin

Use the French date format expected by Cygwin CRT which differs from the
one used by MSVC.

See #23782.

(cherry picked from commit 15d7aaaa50c0e815a3d3b94dece365eaffe6653c)

Kvaz1r committed:
Fix creating wxRadioBox with no items in wxUniv

Don't select the first item if there are no items at all.

See #23784.

(cherry picked from commit d282cc96579ff36f00b4c2f1c9aea3cdca9dfa64)

PBfordev committed:
Improve wxFileSystem::AddHandler() documentation

The note in the method description incorrectly stated that
"...the handlers are deleted in wxFileSystem's destructor so
that you don't have to care about it." However, unless a
handler is removed by calling wxFileSystem::RemoveHandler()
(or undocumented wxFileSystem::CleanUpHandlers()), it is deleted
only when wxWidgets shuts down in wxFileSystemModule::OnExit().

Update the method documentation that it is correct (and similar to
that of wxImage::AddHandler()).

See #23789.

(cherry picked from commit af453674b66f79048f05f934559618c9d2522570)

PBfordev committed:
Fix typo in wxColourDialog documentation

The code example in wxColourDialog documentation used
a wrong type to create wxColourDialog instance.

See #23793, #23794.

(cherry picked from commit 547d49f68bc9241f693e85a4731689ee6d2ac189)

vadz committed:
Improve error reporting for IO errors in wxrc

Notably, give a clear error if an input file is not found and also check
that the temporary output file could have been written successfully.

Still continue handling the other files even if some of them couldn't be
found in order to give all errors at once if there is more than one
missing file.

(cherry picked from commit 13d8adb4e14c38dd7e8531f263d778dfbb0a0630)

vadz committed:
Add support for multi-resolution bitmaps to wxrc

Handle multiple semicolon-separated paths in <bitmap> elements which are
used to provide several versions of the same bitmap in different
resolutions.

This commit is best viewed ignoring whitespace-only changes.

See #23795.

(cherry picked from commit ab88fecdc568857e1ca0f00529dcd92dc71a164c)

vadz committed:
Pin httpbin to the previously working 0.7.0 version

The recently released 0.10.0 doesn't seem to have any actual changes but
has new dependencies which broke the previously working builds.

See #23800.

(cherry picked from commit 55c704a1b404bb302f2fb0a8d9589ab841fcfc92)

vadz committed:
Add MSVS 2022 solution file for wxrc too

Add another manually created solution file, after adding the solutions
for MSVS 2017 and 2019 in cfe4a10995 (Add manually created MSVS 201x
solutions files for wxrc, 2019-01-07) to make it possible to build wxrc
with MSVS 2022 easily too.

(cherry picked from commit 9e8ab658fe87f621b3a5c96e4df43c588d6fd23b)

sthibaul committed:
Fix DynamicLibrary::Load() test for less common platforms

alpha and ia64 use libc.so.6.1, GNU/kFreeBSD uses libc.so.0.1, and
GNU/Hurd uses libc.so.0.3, so check for these versions too in addition
to 6 and 7 used under x86 Linux.

See #23801.

(cherry picked from commit 62f75c235db169bf57beaa61694b3c5db95e5e60)

swt2c commented:
3ae3a4e1f519741132c51e5aa8f83d6f05174501 is probably a candidate too (pcor merged that while you were on vacation).

paulcor committed:
Avoid division-by-zero in wxImage::Paste()

See #23791.

(cherry picked from commit 496a9a40a1853376d877a017a2b6b79b6d6b5f71)

mcorino committed:
Don't stop wxWindowDestroyEvent processing in wxAuiManager

The existing code prevented the other handlers from processing this even
for no good reason, so don't do it any more.

See #23811, #23815.

(cherry picked from commit cd8a854dce8098d27e9b5c6f8d4a731717728482)

unxed committed:
Fix key codes in key up/down events with non-US keyboard layout

Changes using XKB for getting key codes for the events in wxGTK resulted
in always using the key codes for the US keyboard which was incompatible
and arguably wrong too.

Change the logic to only use XKB when we don't have a valid key code in
the current layout to restore the old behaviour.

See #23680, #23818.

(cherry picked from commit 3492bacde5c1985257ea1e85e506093dadc12efa)

Kumazuma committed:
Fix EVT_PRESS_AND_TAP() event macro definition

Use correct type for the event handler cast.

See #23819.

(cherry picked from commit b8d49138cc782e2dad242322acf40f54535b1cf4)

vadz committed:
Fix crash on DPI change for wxStaticBox without a control label

Check that we have a label before repositioning it fix the crash
introduced by 54684be79d (Fix position of label window in wxStaticBox
after DPI change, 2023-08-05).

See #23740, #23826.

(cherry picked from commit 329d35b922aa893c292d14061fb099aacb4af20b)

vadz commented:
Yes, I had already cherry-picked it into my local branch, pushed it now to update the PR.

vadz commented:
As there have been no comments, merging now.
